Source/rclUE/Public/rclcUtilities.h

```diff
@@ -161,6 +161,82 @@ static const TMap<TEnumAsByte<UROS2QoS>, rmw_qos_profile_t> QoS_LUT = {
     {UROS2QoS::System, rmw_qos_profile_system_default},
     {UROS2QoS::UnknownQoS, rmw_qos_profile_unknown}};
 
+UCLASS()
+class URRTimerManager : public UObject
+{
+    GENERATED_BODY()
+protected:
+    UPROPERTY()
+    bool bEnabled = true;
+
+    //! Timer handler for periodic publisher
+    UPROPERTY(VisibleAnywhere, BlueprintReadOnly)
+    FTimerHandle TimerHandle;
+
+    FTimerDelegate Delegate;
+
+    FTimerDelegate TimerDelegate;
+
+    UWorld* World;
+
+    UPROPERTY()
+    float Rate = 0.f;
+
+    UPROPERTY()
+    float desiredTime = 0.f;
+
+public:
+    void StopTimer()
+    {
+        bEnabled = false;
+    }
+
+    void SetTimer(FTimerDelegate const& InDelegate, float InRate, UWorld* InWorld)
+    {
+        Rate = InRate;
+        World = InWorld;
+        Delegate = InDelegate;
+        bEnabled = true;
+        desiredTime = UGameplayStatics::GetTimeSeconds(World) + Rate;
+        // TimerDelegate = FTimerDelegate::CreateUObject(this, &URRTimerManager::SetTimerImple);
+        World->GetTimerManager().SetTimer(TimerHandle, FTimerDelegate::CreateLambda([this] { SetTimerImple(); }), Rate, false);
+    }
+    void SetTimerImple()
+    {
+        if (!bEnabled)
+        {
+            bEnabled = true;
+            return;
+        }
+
+        // function call
+        if (Delegate.IsBound())
+        {
+            Delegate.ExecuteIfBound();
+        }
+        else
+        {
+            UE_LOG(LogTemp, Error, TEXT("Delegate is not bound"));
+        }
+
+        // update desiredTime
+        desiredTime += Rate;
+
+        float wt = desiredTime - UGameplayStatics::GetTimeSeconds(World);
+        while (wt <= 0)
+        {
+            wt += Rate;
+            desiredTime += Rate;
+        }
+
+        UE_LOG(LogTemp, Error, TEXT("SetTimerImpl %f, %f, %f"), desiredTime, Rate, wt);
+
+        // define lambda
+        // TimerDelegate = FTimerDelegate::CreateUObject(this, &URRTimerManager::SetTimerImple);
+        World->GetTimerManager().SetTimer(TimerHandle, FTimerDelegate::CreateLambda([this] { SetTimerImple(); }), wt, false);
```

Comment:
Some forum posts mention lambdas as potential problems with delegates and garbage collection:
https://forums.unrealengine.com/t/dynamic-multicast-delegate-becomes-unbound-after-indeterminate-amount-of-time/127269/7
https://forums.unrealengine.com/t/delegate-gets-unbound-when-captured-by-lambda/379362